docs/.vitepress/config.ts

```diff
@@ -468,16 +443,34 @@ export default defineConfig({
       level: [2, 3],
     },
   },
-  transformPageData(pageData) {
-    const canonicalUrl = `${ogUrl}/${pageData.relativePath}`
-      .replace(/\/index\.md$/, '/')
-      .replace(/\.md$/, '')
-    pageData.frontmatter.head ??= []
-    pageData.frontmatter.head.unshift(
-      ['link', { rel: 'canonical', href: canonicalUrl }],
-      ['meta', { property: 'og:title', content: pageData.title }],
-    )
-    return pageData
+  transformHead(ctx) {
```

Comment:
`transformPageData` was previously use for reasons mentioned at https://github.com/vitejs/vite/pull/15984. However, I can't seem to replicate the issue now. We're only generating one `.html` file which servers would use for both `/path` and `/path.html`. There's no way that canonical links were only added to `/path.html` unless we generate two files or it was SSR.

`transformHead` seems to work the same from my testing and is the more direct API.